fink_client/consumer.py

```diff
@@ -0,0 +1,172 @@
+
+    def __exit__(self, type, value, traceback):
+        self._consumer.close()
+
+    def poll(self, timeout: float = -1) -> (str, dict):
+        """Consume messages from Fink server
+
+        Parameters
+        ----------
+        timeout: float
+            maximum time to block waiting for a message
+            if not set default is None i.e. wait indefinitely
+
+        Returns
+        ----------
+        (topic, alert): tuple(str, dict)
+            returns (None, None) on timeout
+        """
+        msg = self._consumer.poll(timeout)
+        if msg is None:
+            return None, None
+
+        topic = msg.topic()
+        avro_alert = io.BytesIO(msg.value())
+        alert = _decode_avro_alert(avro_alert, self._parsed_schema)
+
+        return topic, alert
+
+    def consume(self, num_alerts: int = 1, timeout: float = -1) -> list:
+        """Consume and return list of messages
+
+        Parameters
+        ----------
+        num_messages: int
+            maximum number of messages to return
+
+        timeout: float
+            maximum time to block waiting for messages
+            if not set default is None i.e. wait indefinitely
+
+        Returns
+        ----------
+        list: [tuple(str, dict)]
+            list of topic, alert
+            returns an empty list on timeout
+        """
+        alerts = []
+        msg_list = self._consumer.consume(num_alerts, timeout)
+
+        for msg in msg_list:
+            topic = msg.topic()
+            avro_alert = io.BytesIO(msg.value())
+            alert = _decode_avro_alert(avro_alert, self._parsed_schema)
+
+            alerts.append((topic, alert))
+
+        return alerts
+
+    def close(self):
+        """Close connection to Fink broker"""
+        self._consumer.close()
+
+
+def _get_kafka_config(servers, config):
+
+    kafka_config = {}
+    default_config = {
+        "bootstrap.servers": "{}".format(",".join(servers)),
+        "auto.offset.reset": "earliest",
+    }
+    
+    if 'username' in config and 'password' in config:
+        kafka_config["security.protocol"] = "sasl_plaintext"
+        kafka_config["sasl.mechanism"] = "SCRAM-SHA-512"
+        kafka_config["sasl.username"] = config["username"]
+        kafka_config["sasl.password"] = config["password"]
+    
+    kafka_config["group.id"] = config["group_id"]
+    
+    kafka_config.update(default_config)
+    
+    # overwrite servers if given (used in tests)
+    if 'bootstrap.servers' in config:
+        kafka_config["bootstrap.servers"] = config["bootstrap.servers"]
+
+    return kafka_config
+
+
+def _get_alert_schema(schema_path=None):
+    if schema_path is None:
+        # get schema from fink-broker
+        try:
+            print("Getting schema from fink servers...")
+            schema_url = "https://raw.github.com/astrolabsoftware/fink-broker/master/schemas/distribution_schema.avsc"
+            r = requests.get(schema_url)
+            os.makedirs("schema", exist_ok=True)
```

Comment:
In `fink-broker`, we use `schemas/` (with `s`). I think to avoid confusion, we should follow the same convention.

Reply:
Thanks for the tip
Fixed in f489afb

Reply:
Thinking a bit more on this - I would suggest the following:
- all schemas must go in the folder `schemas/`.
- In this folder, there is the default one (`fink_alert_schema.avsc`)
- When getting schema from fink servers, the file must be downloaded in `schemas/` as well but keep the same name as in fink-broker to avoid confusion with the default fink-client file.

Namely (not properly checked):

```python
if schema_path is None:
  # get schema from fink-broker
  try:
    print("Getting schema from fink servers...")
    schema_url = "https://raw.github.com/astrolabsoftware/fink-broker/master/schemas/distribution_schema.avsc"
    filename = schema_url.split("/")[-1]
    r = requests.get(schema_url, timeout=1)
    schema_path = os.path.abspath(os.path.join(
      os.path.dirname(__file__), '../schemas/{}'.format(filename)))
    with open(schema_path, "w") as f:
      f.write(r.text)
  except RequestException:
    schema_path = os.path.abspath(os.path.join(
      os.path.dirname(__file__), '../schemas/fink_alert_schema.avsc'))
    m = ("Could not obtain schema from fink servers\n"
        "Using default schema available at: {}").format(schema_path)
    print(m)

```

Reply:
Thanks @JulienPeloton 
This makes much more sense. 
Done in 9b3700f

Reply:
Thanks!